test/plot-schema.json

```diff
@@ -3573,6 +3573,91 @@
       "nonzero"
      ]
     },
+    "label": {
+     "editType": "none",
+     "font": {
+      "color": {
+       "editType": "none",
+       "valType": "color"
+      },
+      "description": "Sets the shape label text font.",
+      "editType": "none",
+      "family": {
+       "description": "HTML font family - the typeface that will be applied by the web browser. The web browser will only be able to apply a font if it is available on the system which it operates. Provide multiple font families, separated by commas, to indicate the preference in which to apply fonts if they aren't available on the system. The Chart Studio Cloud (at https://chart-studio.plotly.com or on-premise) generates images on a server, where only a select number of fonts are installed and supported. These include *Arial*, *Balto*, *Courier New*, *Droid Sans*,, *Droid Serif*, *Droid Sans Mono*, *Gravitas One*, *Old Standard TT*, *Open Sans*, *Overpass*, *PT Sans Narrow*, *Raleway*, *Times New Roman*.",
+       "editType": "none",
+       "noBlank": true,
+       "strict": true,
+       "valType": "string"
+      },
+      "role": "object",
+      "size": {
+       "editType": "none",
+       "min": 1,
+       "valType": "number"
+      }
+     },
+     "padding": {
+      "description": "Sets padding between edge of label and xanchor / yanchor.",
+      "dflt": 3,
+      "editType": "none",
+      "valType": "number"
+     },
+     "role": "object",
+     "text": {
+      "description": "Sets the text to display with shape.",
+      "dflt": "",
+      "editType": "none",
+      "valType": "string"
+     },
+     "textangle": {
+      "description": "Sets the angle at which the label text is drawn with respect to the horizontal. For lines, the default textangle is `auto`, which displays the text at the same angle as the line. For all other shapes, the default textangle is 0 (horizontal).",
+      "editType": "none",
+      "valType": "angle"
+     },
+     "textposition": {
+      "description": "Sets the position of the label text relative to the shape.",
+      "editType": "none",
+      "valType": "enumerated",
+      "values": [
+       "top left",
+       "top center",
+       "top right",
+       "middle left",
+       "middle center",
+       "middle right",
+       "bottom left",
+       "bottom center",
+       "bottom right",
+       "start",
+       "middle",
+       "end"
+      ]
+     },
+     "xanchor": {
```

Comment:
@emilykl, I started working on some examples that demonstrate the `xanchor` and `yanchor`. For `xanchor`, this is what I see for "left" and "right". Is this the correct behaviour where with "left" the label is further to the right (and the opposite behaviour for "right")?  If so, could you help me understand what setting these is doing and what would be a good reason for using them?

<img width="599" alt="image" src="https://user-images.githubusercontent.com/12749831/222214270-2b0a1128-c9a9-4810-9bc5-81e70ce0713c.png">

Reply:
Hi @LiamConnors , good question -- the behavior of `xanchor` and `yanchor` for shape labels is the same as `xanchor` and `yanchor` for [annotations](https://plotly.com/javascript/reference/layout/annotations/).

The docstrings I added are not very informative though. It probably makes sense to use almost the same docstrings as for annotations, which has a more complete explanation.

Reply:
This is correct, and it's true that it's a bit confusing but it's consistent with annotations and several other items. `xanchor` means "which side of the item is anchored to the specified position?" So in your example you must have `textposition='middle center'`, therefore `xanchor: 'right'` means the right edge of the text is at the middle of the shape.

When the position is `'middle center'` there won't be tons of reasons to use x/yanchors that aren't also middle & center. But imagine two things you might want to do on the edge of a box:
```
+------------+
|     Label 1|Label 2
|            |
|            |
+------------+
```
Both of these have `textposition: 'top right'` and both have `yanchor: 'top'` (the top edge of the text is anchored to the textposition), but Label 1 has its right edge anchored to the text position (`xanchor: 'right'`) whereas Label 2 has its left edge anchored to the text position (`xanchor: 'left'`). You might also want the text on top of the box, aligned to its right edge; that would be the same as Label 1 but with `yanchor: 'bottom'`.